client/src/app/components/search/search.component.ts

```diff
@@ -70,22 +70,19 @@ export class SearchComponent implements OnInit {
     }
 
     public onFilterSubmit() {
-        const params = this.filterForm.value;
-        console.log(this.filterForm.value)
-        this.loaderService.setLoading(true);
-        this.taskService.getFilteredTasks(params).pipe(
-            finalize(() => {
-                this.loaderService.setLoading(false);
-            }),
-            takeUntil(this.ngUnsubscribe)
-        ).subscribe(
-            tasks => {
-                console.log(tasks)
+        const params = {};
+        const keys = Object.keys(this.filterForm.value);
+        keys.forEach(key=>{
+            if(this.filterForm.value[key]){
+                params[key] = this.filterForm.value[key];
+            }
+        })
+        const tagParam = this.getTagParam();
+        if(tagParam){
+            params['tags'] = tagParam;
+        }
```

Comment:
If you want to do it more generic, you can check if value is an array. Additionally you can simplify it and use `Object.entries`, like so:

```
          Object
            .entries(this.filterForm.value)
            .forEach(([ key, value ]) => {
                if (!value) {
                    return
                }

                if (Array.isArray(value)) {
                    // ....

                    return
                }

                params[key] = value;
            })
```

Pay attention, that in this example, first I'm checking if we can immidiately return. If you do opposite condition, like `if(value) ...` then you will have extra tab indent for usually larger code block, which negatively impact code readability.

Reply:
Done and if value is an array we just do
```
                   const arrayString = value.join();
                    params[key] = arrayString;
                    return;
```